src/core/libraries/kernel/memory_management.cpp

```diff
@@ -207,6 +208,19 @@ int PS4_SYSV_ABI sceKernelQueryMemoryProtection(void* addr, void** start, void**
     return memory->QueryProtection(std::bit_cast<VAddr>(addr), start, end, prot);
 }
 
+int PS4_SYSV_ABI sceKernelMProtect(const void* addr, size_t size, int prot) {
+    Core::MemoryManager* memory_manager = Core::Memory::Instance();
+    Core::MemoryProt protection_flags = static_cast<Core::MemoryProt>(prot);
+    return memory_manager->Protect(std::bit_cast<VAddr>(addr), size, protection_flags);
```

Comment:
So the equivalent to memory_manager->Protect here modifies g_posix_errno. In shad, that would be a call to SetPosixErrno. As for the __error call, that corresponds to __Error in shadPS4's libkernel.cpp. When using SetPosixErrno, make sure the errno value being set is accounted for in SetPosixErrno, as the expected errno value is different between OSes.

Reply:
sceKernelMProtect is not a posix function dunno what that should done

Reply:
Based on a decomp, it seems like sceKernelMProtect calls some mprotect function (with proper NID and all). Is that function a posix function?

Reply:
mprotect is a posix function

Reply:
most scekernel function calls posix ones

Reply:
If this https://github.com/idc/ps4libdoc/blob/9.00/system/common/lib/libkernel.sprx.json is anything to go by, the mprotect function libkernel uses is a posix function, so that's probably why errno is set.